### PR TITLE
Review API endpoint and tests

### PR DESCRIPTION
**Pull request by hwbit**
New Review endpoint and tests
Route: `/api/reviews`

Changed model name from `comment` to `review`.

### CODE REVIEW

File: preppal-be/routes/reviewApi.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+const Review = require("../models/review.ts");
+const Author = require("../models/user.ts");
+const Recipe = require("../models/recipe.ts");
+
+const routerReviewApi = expressReviewApi.Router();
+
+
+/**
+ * GET - Get all recipes
```
Comment (lik5) [inline preppal-be/routes/reviewApi.ts:12]: get all reviews?

File: preppal-be/routes/reviewApi.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    }
+});
+
+/**
+ * GET - Get reviews by recipeId
+ */
+routerReviewApi.get("/:recipeId", async (req, res) => {
+    try {
+        const reviews = await Review.findOne({ recipeId: req.params.recipeId });
+        res.status(200).json(reviews);
```
Comment (lik5) [inline preppal-be/routes/reviewApi.ts:31]: probably only want to review only the reviews?

no harm in returning the recipe id as well I guess

File: preppal-be/routes/reviewApi.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+ * GET - Init the review for a recipe
+ */
+routerReviewApi.get("/new/:recipeId", async (req, res) => {
+    try {
+        const recipeId = req.params.recipeId;
+
+        const reviews = await Review.findOne({ recipeId });
+        const recipe = await Recipe.findOne({ _id: recipeId });
+
+        if (reviews) {
```
Comment (lik5) [inline preppal-be/routes/reviewApi.ts:49]: does this work? Doesn't this find if reviews exist and if so cannot add more reviews?

File: preppal-be/routes/reviewApi.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            return res.status(400).json({msg: "Comment requires a rating."});
+        } else if (!title) {
+            return res.status(400).json({msg: "Comment requires a title."});
+        }
+
+        const recipeReview = await Review.findOne({ recipeId });
+        const authorReview = await Author.findOne({ username: author });
+
+        if (!recipeReview) {
+            return res.status(400).json({ errors: [{ msg: "Invalid Id for recipe." }] });
```
Comment (lik5) [inline preppal-be/routes/reviewApi.ts:94]: lower case id

File: preppal-be/routes/reviewApi.ts [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        } else if (!authorReview) {
+            return res.status(400).json({ errors: [{ msg: "Invalid Id for author." }] });
```
Comment (lik5) [inline preppal-be/routes/reviewApi.ts:96]: lower case id

File: preppal-be/routes/reviewApi.ts (hunk 3 above)
Comment (lik5) [reply preppal-be/routes/reviewApi.ts:49]: unless this is just creating an entry for recipe reviews in our database, but I don't know if it should send a 400 error. Up to you

File: preppal-be/routes/reviewApi.ts [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+        // adds item to the front of the array and update
+        // this is to get the comments to show up in reverse chronological order (most recent at the top)
+        await reviews.unshift(review);
+        await Review.findByIdAndUpdate(recipeReview.id, { reviews });
+
+        const updatedRecipeReview = await Review.findOne({ recipeId });
+
+        if (!updatedRecipeReview) {
+            return res.status(404).json({ msg: "Review was not found." });
```
Comment (lik5) [inline preppal-be/routes/reviewApi.ts:118]: should probably say save review unsuccessful instead

File: preppal-be/routes/reviewApi.ts (hunk 3 above)
Comment (hwbit) [reply preppal-be/routes/reviewApi.ts:49]: We create a new table in the database for reviews. Each recipe is pretty much only holds an array of reviews.
This endpoint is only called at the same time when we create a new recipe. It was also used to manually create these new tables for older recipes.

So how this works is that it first checks if this table exists for a review. Then it checks to make sure the recipe exists. If those a review doesn't exist and the recipe does, it creates a new table. We do this just so we don't duplicate review sections for each recipe.

**Review by lik5 — APPROVED** [review]

**Pull request merged**